routing/dht/routing.go

```diff
@@ -30,21 +30,21 @@ var asyncQueryBuffer = 10
 
 // PutValue adds value corresponding to given Key.
 // This is the top level "Store" operation of the DHT
-func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte) error {
+func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte, sign bool) error {
```

Comment:
we shouldn't grow this interface. isnt the dht able to tell from the key what validators it falls into? (maybe checking a signature should be validator based).

```
/ca/<hash-of-pk>
```
`ca` for content-addressed. so we can have a validator for `/ca` that knows to hash the value and check it matches the `<hash>` part.